### PR TITLE
Issues/#109 ft merkle sha3

### PR DESCRIPTION
**Pull request by ealefad**
A fix to make these functional tests pass after core Merkle Root SHA3 implementation:
```
feature_block.py 
feature_cltv.py  
feature_csv_activation.py  
feature_dersig.py                
feature_nulldummy.py
mining_basic.py
p2p_compactblocks.py
p2p_invalid_block.py
p2p_segwit.py
p2p_unrequested_blocks.py
wallet_bumpfee.py
```
1. Modified `.travis.yml` to be able to run functional tests with Python3.6 which has updated `hashlib.py` library. `Hashlib v3.6` supports SHA3 algorithms.
1. Only MerkleRoot calculations switched to SHA3_256 in `eqbtest/framework/messages.py`

### CODE REVIEW

File: eqbtest/functional/test_framework/messages.py [hunk 1]
```diff
@@ -579,14 +593,18 @@ def get_merkle_root(cls, hashes):
...
+                # newhashes.append(hash256(hashes[i] + hashes[i2]))
+                newhashes.append(hash3_256(hashes[i] + hashes[i2]))
             hashes = newhashes
         return uint256_from_str(hashes[0])
 
     def calc_merkle_root(self):
         hashes = []
         for tx in self.vtx:
             tx.calc_sha256()
+            # TODO: DON'T Swithch Tx hash calculation to SHA3_256 yet
```
Comment (macterra) [inline eqbtest/functional/test_framework/messages.py]: I think we are converging on a new convention to use EQB_TODO in comments. Plus there's a typo in this one :)

File: eqbtest/functional/test_framework/messages.py [hunk 2]
```diff
@@ -474,6 +480,14 @@ def calc_sha256(self, with_witness=False):
             self.sha256 = uint256_from_str(hash256(self.serialize_without_witness()))
         self.hash = encode(hash256(self.serialize_without_witness())[::-1], 'hex_codec').decode('ascii')
 
+    def calc_sha3_256(self, with_witndess=False):
```
Comment (macterra) [inline eqbtest/functional/test_framework/messages.py]: typo? with_witness

Comment (ealefad) [reply eqbtest/functional/test_framework/messages.py]: Yes, both in the parameter and later in the if statement :) Will correct

File: eqbtest/functional/test_framework/messages.py (hunk 1 above)
Comment (ealefad) [reply eqbtest/functional/test_framework/messages.py]: I was convinced that I saw it somewhere but couldn't find, yes, EQB_TODO

**Review by macterra — APPROVED** [review]

**Pull request merged**